### ISSUE TITLE
Multiple issues with fortios_system_admin module

### ISSUE THREAD

**Issue by JonTheNiceGuy** [fortinet-solutions-cse/ansible_fgt_modules#4]
I'm working on some automation with @tohillm and with the following playbook:

```yaml
---
- hosts: localhost
  gather_facts: false
  vars:
      host1: "192.0.2.1"
      fw_username: "admin_user"
      fw_password: "20190115_Admin"
      user_username: "radius_user"
      user_password: "20190115_User"
      vdom: "root"

  tasks:
  - name: Configure RADIUS server entries.
    fortios_user_radius:
      host:  "{{ host1 }}"
      username: "{{ fw_username }}"
      password: "{{ fw_password }}"
      https: True
      vdom: "{{ vdom }}"
      user_radius:
        state: "present"
        name: "radserv"
        radius-port: "1812"
        secret: "public"
        server: "192.0.2.2"
        source-ip: "192.0.2.1"
        timeout: "45"
        auth-type: "auto"

  - name: Configure auth group
    fortios_user_group:
      host:  "{{ host1 }}"
      username: "{{ fw_username }}"
      password: "{{ fw_password }}"
      https: True
      vdom: "{{ vdom }}"
      user_group:
        state: present
        group-type: firewall
        name: admin-radius
        member:   
        - name: NPS4

  - name: Configure admin users.
    fortios_system_admin:
      host:  "{{ host1 }}"
      username: "{{ fw_username }}"
      password: "{{ fw_password }}"
      https: True
      vdom: "{{ vdom }}"
      system_admin:
        state: "present"
        accprofile: "super_admin"
        name: "{{ user_username }}"
        password: "{{ user_password }}"
        vdom:
        - name: "{{ vdom }}"
        remote-auth: "enable"
        remote-group: "admin-radius"
        wildcard: enable
```

When I run this playbook, and I've copied `fortios_user_radius.py`, `fortios_user_group.py` and `fortios_system_admin.py` into `library/`, I get:

```
PLAY [localhost] **************************************************************************************************************************************************************************************************

TASK [Configure RADIUS server entries.] ***************************************************************************************************************************************************************************
changed: [localhost]

TASK [Configure auth group] ***************************************************************************************************************************************************************************************
changed: [localhost]

TASK [Configure admin users.] *************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "implementation error: unknown type password-2 requested for password"}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=2    unreachable=0    failed=1
```

If I edit `library/fortios_system_admin.py` and update (line 884)[https://github.com/fortinet-solutions-cse/ansible_fgt_modules/blob/c739655aed2bb1de27065b4ba3bc2c8c6de82f06/v6.0.2/system/fortios_system_admin.py#L844] to show:

```
"password": {"required": False, "type": "str"},
```

I then get the following:

```
PLAY [localhost] **************************************************************************************************************************************************************************************************

TASK [Configure RADIUS server entries.] ***************************************************************************************************************************************************************************
changed: [localhost]

TASK [Configure auth group] ***************************************************************************************************************************************************************************************
changed: [localhost]

TASK [Configure admin users.] *************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "meta": {"build": 200, "http_method": "PUT", "http_status": 403, "mkey": "radius_user", "name": "admin", "path": "system", "revision": "194.0.0.9544640475189144756.1547457824", "serial": "FGVM04TMxxxxxxx", "status": "error", "vdom": "root", "version": "v6.0.3"}, "msg": "Error in repo"}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=2    unreachable=0    failed=1   
```

It does complete successfully if I remove these lines from the last task:

```
        vdom:
        - name: "{{ vdom }}"
        remote-auth: "enable"
        remote-group: "admin-radius"
        wildcard: enable
```

I see two issues here:

1. `password` is defined as `password-2` which is not a known type - is this something internal that you need to share? 😄 Or, perhaps, this needs to be changed to `str`?
2. For some reason, one of these lines in the last block above prevents the user from being added. Please can you advise?

**Comment by mamunozgonzalez** [comment]
Hi @JonTheNiceGuy ,
  yes, there seems to be an error with that module. Actually it has not been validated yet (it's pending to be submitted to Ansible). 
I will have a look to it. I'll keep you posted.

Thanks for pointing this out. This is really good help!

**Comment by JonTheNiceGuy** [comment]
Glad to be of help.

**Comment by tohillm** [comment]
Here's the schema we tries to run Ansible Generator and had some issues running it:
```
tohillm@localhost ansible_generator-master]$ 
[tohillm@localhost ansible_generator-master]$ ./generate.py 
Traceback (most recent call last):
  File "./generate.py", line 100, in <module>
    fgt_schema = json.loads(fgt_schema_file)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Unterminated string starting at: line 47566 column 24 (char 1597420)
[tohillm@localhost ansible_generator-master]$ 
```
In the interim, here is the schema output:
https://gist.github.com/tohillm/357c6d028d3969359f375a037701d747

I hope this helps.

**Comment by mamunozgonzalez** [comment]
Hi @tohillm,
 I downloaded your schema for 6.0.3, rename it to fgt_schema.json and put it in the same dir as where 'generate.py' is, and it runs fine. Actually it generates 429 modules. 

Can you ensure your fgt_schema.json is complete? I downloaded your example with this command:
`wget https://gist.githubusercontent.com/tohillm/357c6d028d3969359f375a037701d747/raw/343b6898ccc22885c19034d0e6494b5547478142/6.0.3.json`

**Comment by JonTheNiceGuy** [comment]
I can confirm, the fgt_schema.json was incomplete - I'm parsing the output differences in my spare time between the 6.0.3 we generated and the 6.0.2. The 6.0.3 schema is at https://gist.github.com/JonTheNiceGuy/77f97ca0b7bff82431421e90bc8484b6

Looks like the Schema from a cloud model (this was from an Azure 6.0.3 image) has options that the Physical box doesn't (e.g. your 6.0.2 has stuff about USB sticks, my 6.0.3 has stuff about autoscaling).

It's probably worth adding a requirements.txt to the ansible_generator project, as I certainly had issues running it with my Ubuntu-WSL session (I think I ended up `apt install`ing `python-jinja2` and `python-autopep8` to make it work) but that's not important right now.

I now need to run this module against my FGT and see if it's fixed the other part of the issue above.
